examples/core/dof/app.js

```diff
@@ -0,0 +1,433 @@
+          },
+          indices: indexBuffer
+        },
+        uniforms: {
+          uTexture: textures[0]
+        },
+        instanced: true,
+        instanceCount: NUM_BOXES,
+        drawMode: GL.TRIANGLES,
+        vertexCount: cubeGeo.indices.value.length
+      });
+
+      const quadVertexArray = new VertexArray(gl, {
+        program: dofProgram,
+        attributes: {
+          aPosition: new Buffer(gl, new Float32Array(QUAD_VERTS))
+        }
+      });
+
+      return {
+        instancedBoxes,
+        boxXforms,
+        boxMatrices,
+        matrixBuffer,
+        mainFramebuffer,
+        dofFramebuffer,
+        quadVertexArray,
+        dofProgram
+      };
+    });
+  },
+
+  onRender: ({gl, tick, width, height, aspect, instancedBoxes, boxMatrices, boxXforms, matrixBuffer, mainFramebuffer, dofFramebuffer, quadVertexArray, dofProgram}) => {
+
+    mainFramebuffer.resize(gl.drawingBufferWidth, gl.drawingBufferHeight);
+    dofFramebuffer.resize(gl.drawingBufferWidth, gl.drawingBufferHeight);
+
+    clear(gl, {color: [0, 0, 0, 1], depth: true, framebuffer: mainFramebuffer});
+
+    const camView = new Matrix4().lookAt({eye: [3, 1.5, 3], center: [0, 0, 0], up: [0, 1, 0]});
+    const camProj = new Matrix4().perspective({fov: radians(75), aspect, near: 0.1, far: 30});
+
+    ////////////////////////////////////////
+    // Update model matrix data and then
+    // update the attribute buffer
+    ////////////////////////////////////////
+
+    for (let i = 0; i < NUM_BOXES; ++i) {
+      let box = boxXforms[i];
+      box.rotate[0] += 0.01;
+      box.rotate[1] += 0.02;
+      box.matrix.identity().translate(box.translate).rotateXYZ(box.rotate).scale(box.scale);
+      boxMatrices.set(box.matrix, i * 16);
+    }
+
+    matrixBuffer.setData(boxMatrices);
+
+    ////////////////////////////////////
+    // Draw boxes to main framebuffer
+    ////////////////////////////////////
+
+    instancedBoxes.draw({
+      uniforms: {
+        uView: camView,
+        uProjection: camProj
+      },
+      framebuffer: mainFramebuffer
+    });
+
+    /////////////////
+    // Apply DOF
+    /////////////////
+
+    // Horizontal DOF blur
+    clear(gl, {color: [0, 0, 0, 1], framebuffer: dofFramebuffer});
+
+    // texelOffset determines the direction of the blur
+    texelOffset[0] = 1;
+    texelOffset[1] = 0;
+
+    dofProgram.setUniforms({
+        uTexelOffset: texelOffset,
+        uColor: mainFramebuffer.color,
+        uDepth: mainFramebuffer.depth
+    });
+
+    dofProgram.draw({
+      vertexArray: quadVertexArray,
+      drawMode: gl.TRIANGLE_STRIP,
+      vertexCount: 4,
+      framebuffer: dofFramebuffer
+    });
+
+    // Vertical DOF blur
+    clear(gl, {color: [0, 0, 0, 1]});
+
+    texelOffset[0] = 0;
+    texelOffset[1] = 1;
+
+    dofProgram.setUniforms({
+        uTexelOffset: texelOffset,
+        uColor: mainFramebuffer.color,
+        uDepth: mainFramebuffer.depth
+    });
+
+    dofProgram.draw({
```

Comment:
Nit: wrap this into a model object?

Reply:
Conceptually, I find it a bit awkward to think of a post processing screen-space pass as a "model". If it's really a break from Luma programming conventions, though, I can change it.